ui/ui-components/components/alerts/Toast.tsx

```diff
@@ -0,0 +1,62 @@
+import { useState, useEffect } from "react";
+import { Toast } from "react-bootstrap";
+import { ErrorHandler } from "../../utils/errorHandler";
+import { SuccessHandler } from "../../utils/successHandler";
+
+const secondsToShow = 1000 * 4; // show toasts for 4 seconds before hiding
+const subscriptionKey = "grouparoo-toast";
+
+export default function GrouparooToast({
+  handler,
+  variant,
+}: {
+  handler: SuccessHandler | ErrorHandler;
+  variant: "success" | "danger";
+}) {
+  const [message, setMessage] = useState("");
+  const [show, setShow] = useState(false);
+
+  useEffect(() => {
+    handler.subscribe(subscriptionKey, subscription.bind(this));
+
+    return () => {
+      handler.unsubscribe(subscriptionKey);
+    };
+  }, []);
+
+  function subscription({ message }) {
+    setMessage(formatMessage(message));
+    setShow(true);
+  }
+
+  function formatMessage(message: unknown) {
+    let formattedMessage = String(message);
```

Comment:
I'm concerned that we don't know what the message is. Casting as String may throw an error at some point. Is there a way the type of message can be more specific?